### ISSUE TITLE
 AttributeError: 'GCENode' object has no attribute 'prometheus_data_dir'

### ISSUE THREAD

**Issue by kongove** [scylladb/scylla-cluster-tests#225]
Try to download prometheus data before prometheus setup, then it's failed.

```
2017-01-05 10:12:17,343 stacktrace       L0039 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:422
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR| Traceback (most recent call last):
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|   File "sdcm/tester.py", line 131, in wrapper
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|     args[0].clean_resources()
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|   File "sdcm/tester.py", line 760, in clean_resources
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|     self.monitors.destroy()
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|   File "sdcm/cluster.py", line 3138, in destroy
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|     self.download_monitor_data()
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|   File "sdcm/cluster.py", line 1778, in download_monitor_data
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|     node.download_prometheus_data_dir()
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|   File "sdcm/cluster.py", line 534, in download_prometheus_data_dir
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR|     (self.remoter.user, self.remoter.user, self.prometheus_data_dir))
2017-01-05 10:12:17,343 stacktrace       L0042 ERROR| AttributeError: 'GCENode' object has no attribute 'prometheus_data_dir'
```

**Comment by lmr** [comment]
This happens due to the  cleanup stage.

**Comment by lmr** [comment]
This happens usually when the setup code itself fails, which is rare so it wasn't accounted for. We can make the cleanup code more robust to not fail in such occasions.

**Comment by lmr** [comment]
See also #227

**Comment by kongove** [comment]
I had rebased my patch, will send the PR later.

**Comment by kongove** [comment]
solved by: commit 2074ae49a19c3e5c0bd4f6a822fe2ee177f78f1a
